packages/volto/src/components/manage/Widgets/ColorPickerWidget.tsx

```diff
@@ -44,21 +69,29 @@ const ColorPickerWidget = (props) => {
 
               <div className="buttons">
                 {colors.map((color) => {
+                  let colorValue: string | Color['style'];
+                  const colorName = color.name;
+                  if (color.style !== undefined) {
+                    colorValue = color.style;
+                  } else {
+                    colorValue = color.name;
+                  }
                   return (
                     <Button
-                      key={id + color.name}
-                      className={color.name}
+                      key={id + colorName}
```

Comment:
I think we could just use the index in `colors` here and avoid requiring `name`. It seems unlikely that they will be reordered frequently.

Reply:
@davisagli so you'd change the `style` use case branch? as in:

```ts
type Color =
  | {
      name: undefined;
      label: string;
      style: Record<`--${string}`, string>;
    }
  | {
      name: string;
      label: string;
      style: undefined;
    };
```

Reply:
@sneridagh yeah, that's the suggestion, but I don't feel strongly about it

Reply:
@davisagli I don't know, I feel like this is more consistent, when we deal with choices, we always have a "display name" (label) and a "token" (value). The token is invariable, but the display name could be i18n'd. If at some point you need to identify the color entry, you will have to equal the style property (which you could still do...) Answering myself while writing :) 

Anyways, I don't have a strong opinion either at this point. 

/cc @tiberiuichim @ichim-david @giuliaghisini @pnicolli any opinions about this?